### PR TITLE
{ARM} Fix the bug for live test

### PR DESCRIPTION
**Pull request by zhoxing-ms**
**Description**  

Fix the bug for live test:

***1. Fixed resource names***
* Errors:
   - test_subscription_level_deployment               *The storage account named armbuilddemo18221 is already taken*
   - test_subscription_level_deployment_old_command   *The storage account named armbuilddemo18221 is already taken.*
   - test_group_deployment_with_large_params   *cli_test_deployment_with_large_paramsxxx is not valid.*
* Solutions:
   - Use the random method to change the resource name to a random value

***2. The subscription ID belonging to service team is used***
* Errors:
   - test_management_group_deployment  *knack.util.CLIError: Azure Error: CrossTenantDeploymentNotPermitted*
   - test_management_group_level_what_if  *knack.util.CLIError: Azure Error: CrossTenantDeploymentNotPermitted*
   - test_resource_policy_management_group  Reader level access is required on the management group or group ancestors.
* Solutions:
   - Change the subscription ID to our own

***3. The check status value is not correct***
* Errors:
   - test_feature_unregister  *'properties.state' doesn't yield expected value 'Unregistered', instead the actual value is 'Unregistering'*
* Solutions:
    - Change to the correct check value

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 1]
```diff
@@ -1437,7 +1441,7 @@ def test_feature_list(self):
 
     @AllowLargeResponse(8192)
     def test_feature_unregister(self):
-        self.cmd('feature unregister --namespace Microsoft.Network --name AllowLBPreview', checks=self.check('properties.state', 'Unregistered'))
+        self.cmd('feature unregister --namespace Microsoft.Network --name AllowLBPreview', checks=self.check('properties.state', 'Unregistering'))
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1444]: Is it possible that it is finished? Can we check Unregistered or Unregistering?

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1444]: Yes, it is possible~ Actually, I sent an email to the service team to ask the expected status before, but I haven't received any reply yet.
If I receive a reply, I will synchronize the information in this PR.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1444]: Maybe regular expression works

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**